manifests/windows.pp

```diff
@@ -65,11 +66,14 @@
       $ensure_version = $agent_version
     }
 
+    $hostname_option = $hostname ? { '' => {}, default => { 'HOSTNAME' => $hostname } }
```

Comment:
Why do we need to change this? It didn't work if you didn't specify the hostname?

Reply:
It didn't work, the installer ended up receiving `HOSTNAME="NPM=TRUE"`. Probably the same could happen if you leave the API key empty, but I didn't investigate that (and it's not as common to leave it empty). This shouldn't be the case with the tags, because even if they are empty we pass `TAGS=""`.